src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/GetDownloadResultUtility.cs

```diff
@@ -51,12 +59,25 @@ public static class GetDownloadResultUtility
                                 return new DownloadResourceResult(DownloadResourceResultStatus.NotFound);
                             }
 
-                            return await GlobalPackagesFolderUtility.AddPackageAsync(
+                            if (cacheContext.NoCache)
+                            {
+                                return await AddPackageDirectAsync(
```

Comment:
Is there a lot of similar code between `GlobalPackagesFolderUtility.AddPackageAsync` and `AddPackageDirectAsync`? If so, they should share that code.